cmd/agent/api/grpc.go

```diff
@@ -0,0 +1,46 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+/*
+Package api implements the agent IPC api. Using HTTP
+calls, it's possible to communicate with the agent,
+sending commands and receiving infos.
+*/
+package api
+
+import (
+	"context"
+
+	pb "github.com/DataDog/datadog-agent/cmd/agent/api/pb"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	hostutil "github.com/DataDog/datadog-agent/pkg/util"
+)
+
+type server struct {
+	pb.UnimplementedAgentServer
+}
+
+type serverSecure struct {
+	pb.UnimplementedAgentServer
+}
+
+func (s *server) GetHostname(ctx context.Context, in *pb.HostnameRequest) (*pb.HostnameReply, error) {
+	h, err := hostutil.GetHostname()
+	if err != nil {
+		return &pb.HostnameReply{}, err
+	}
+	return &pb.HostnameReply{Hostname: h}, nil
+}
+
+// AuthFuncOverride will override the AuthFunc registered with the unary interceptor
```

Comment:
Unused since you've used a second gRPC server? Or undeclared in the gRPC server?

Reply:
It's not unused, we need to implement the `ServiceAuthFuncOverride` to allow overriding the default `AuthFunc`

https://godoc.org/github.com/grpc-ecosystem/go-grpc-middleware/auth#ServiceAuthFuncOverride 

I will add a comment.